### ISSUE TITLE
Message stored as an array in component history is causing a graphql error and all new versions to fail

### ISSUE THREAD

**Issue by sbland** [teambit/bit#7159]
### Describe the bug
Going to my component page locally or on bit.cloud results in the following error: 
`Error: String cannot represent value: ["async-hooks/use-async-object-manager", "linting"]`

### Steps to Reproduce

1. Go to https://bit.cloud/samnbuk/react_db_client/components/popup-panel-v2

### Expected Behavior

Should show the component page

**Comment by sbland** [comment]
I've even tried bringing the component into a new workspace and I get the same error referring to `async-hooks/use-async-object-manager` which isn't actually a dependent or dependency of this component.

**Comment by sbland** [comment]
Searching "use-async-object-manager" inside the local folder returns 0 results.

**Comment by sbland** [comment]
This seems to be a graphql issue. 
Request:
```
{
  "operationName": "Component",
  "variables": {
    "id": "components/search-and-select",
    "extensionId": "teambit.workspace/workspace"
  },
  "query": "query Component($id: String!, $extensionId: String!, $logType: String, $logOffset: Int, $logLimit: Int, $logHead: String, $logSort: String) {\n  getHost(id: $extensionId) {\n    id\n    get(id: $id) {\n      ...componentFields\n      __typename\n    }\n    __typename\n  }\n}\n\nfragment componentFields on Component {\n  id {\n    ...componentIdFields\n    __typename\n  }\n  ...componentOverviewFields\n  packageName\n  latest\n  compositions {\n    identifier\n    displayName\n    __typename\n  }\n  tags {\n    version\n    __typename\n  }\n  logs(type: $logType, offset: $logOffset, limit: $logLimit, head: $logHead, sort: $logSort) {\n    id\n    message\n    username\n    email\n    date\n    hash\n    tag\n    __typename\n  }\n  __typename\n}\n\nfragment componentIdFields on ComponentID {\n  name\n  version\n  scope\n  __typename\n}\n\nfragment componentOverviewFields on Component {\n  id {\n    ...componentIdFields\n    __typename\n  }\n  aspects(include: [\"teambit.preview/preview\", \"teambit.envs/envs\"]) {\n    id\n    data\n    __typename\n  }\n  elementsUrl\n  description\n  deprecation {\n    isDeprecate\n    newId\n    __typename\n  }\n  labels\n  displayName\n  server {\n    env\n    url\n    __typename\n  }\n  buildStatus\n  env {\n    id\n    icon\n    __typename\n  }\n  size {\n    compressedTotal\n    __typename\n  }\n  preview {\n    includesEnvTemplate\n    legacyHeader\n    isScaling\n    __typename\n  }\n  compositions {\n    identifier\n    displayName\n    __typename\n  }\n  __typename\n}\n"
}
```

Response:
```
{
  "errors": [
    {
      "message": "String cannot represent value: [\"async-hooks/use-async-object-manager\", \"linting\"]",
      "locations": [
        {
          "line": 31,
          "column": 5
        }
      ],
      "path": [
        "getHost",
        "get",
        "logs",
        21,
        "message"
      ],
      "ERR_CODE": "TypeError"
    }
  ],
  "data": {
    "getHost": {
      "id": "react_db_client",
      ....
      "__typename": "ComponentHost"
    }
  }
}
```

**Comment by sbland** [comment]
If I copy all the component code into a brand new component it all works fine in the new component.

**Comment by sbland** [comment]
Full graphql error after running `bit start --log`
```
teambit.harmony/graphql, graphql got an error during running the following query:
    query: "query Component($id: String!, $extensionId: String!, $logType: String, $logOffset: Int, $logLimit: Int, $logHead: String, $logSort: String) {\n  getHost(id: $extensionId) {\n    id\n    get(id: $id) {\n      ...componentFields\n      __typename\n    }\n    __typename\n  }\n}\n\nfragment componentFields on Component {\n  id {\n    ...componentIdFields\n    __typename\n  }\n  ...componentOverviewFields\n  packageName\n  latest\n  compositions {\n    identifier\n    displayName\n    __typename\n  }\n  tags {\n    version\n    __typename\n  }\n  logs(type: $logType, offset: $logOffset, limit: $logLimit, head: $logHead, sort: $logSort) {\n    id\n    message\n    username\n    email\n    date\n    hash\n    tag\n    __typename\n  }\n  __typename\n}\n\nfragment componentIdFields on ComponentID {\n  name\n  version\n  scope\n  __typename\n}\n\nfragment componentOverviewFields on Component {\n  id {\n    ...componentIdFields\n    __typename\n  }\n  aspects(include: [\"teambit.preview/preview\", \"teambit.envs/envs\"]) {\n    id\n    data\n    __typename\n  }\n  elementsUrl\n  description\n  deprecation {\n    isDeprecate\n    newId\n    __typename\n  }\n  labels\n  displayName\n  server {\n    env\n    url\n    __typename\n  }\n  buildStatus\n  env {\n    id\n    icon\n    __typename\n  }\n  size {\n    compressedTotal\n    __typename\n  }\n  preview {\n    includesEnvTemplate\n    legacyHeader\n    isScaling\n    __typename\n  }\n  compositions {\n    identifier\n    displayName\n    __typename\n  }\n  __typename\n}\n"
    variables: {
      "id": "components/search-and-select",
      "extensionId": "teambit.workspace/workspace"
    }
    operationName: "Component"
    raw: false
teambit.harmony/graphql, graphql error 
    TypeError: String cannot represent value: ["async-hooks/use-async-object-manager", "linting"]
        at GraphQLScalarType.serializeString [as serialize] (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/type/scalars.js:159:9)
        at completeLeafValue (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:635:37)
        at completeValue (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:579:12)
        at completeValue (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:557:21)
        at completeValueCatchingError (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:495:19)
        at resolveField (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:435:10)
        at executeFields (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:275:18)
        at collectAndExecuteSubfields (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:713:10)
        at completeObjectValue (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:703:10)
        at completeValue (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:591:12)
        at completeValueCatchingError (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:495:19)
        at /home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:618:25
        at Array.forEach (<anonymous>)
        at forEach (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/iterall/index.js:83:25)
        at completeListValue (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:614:24)
        at completeValue (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:573:12)
        at completeValue (/home/USER/.bvm/versions/0.0.867/bit-0.0.867/node_modules/graphql/execution/execute.js:557:21)
```

**Comment by sbland** [comment]
Tried reverting Bit version to `0.0.744` no change

**Comment by sbland** [comment]
A bvm version issue meant I wasn't using the correct version of Bit. Now using versions below:

```
  bit version   : 0.1.11
  node version  : v16.16.0
  npm version   : 8.1.0
  yarn version  : 1.22.5
  platform      : linux
```

**Comment by sbland** [comment]
Corrupt tag info:
```
tag 0.2.44 (c50029c5d244145e976b73cd9209dd57263b1701)
author: samnbuk 
date: 2023-02-04 21:48:55

      async-hooks/use-async-object-manager,linting
```

**Comment by sbland** [comment]
Changing the line here https://github.com/teambit/bit/blob/624c0ac78b7ae3e0a69c1e86ee6079b7db41984f/scopes/component/component/component.graphql.ts#L47
to `         message: [String]! ` showed that the message is indeed a array...

```
 {
            "id": "db812014fd87312d4c1c02d784cc786b3b87e3c8",
            "message": [
              "async-hooks/use-async-object-manager",
              "linting"
            ],
            "date": "1675547335726"
          },
```

**Comment by sbland** [comment]
I can't see anyway of repairing this. My workaround is now to create a new component and copy the code across.

**Comment by sbland** [comment]
Probably my final update

Comparing the history of 2 components that had been tagged at the same time where one is corrupted(has a message as an array) and another that is fine.

First is corrupted example
```
{
    "id": "db812014fd87312d4c1c02d784cc786b3b87e3c8",
    "message": [
        "async-hooks/use-async-object-manager",
        "linting"
    ],
    "date": "1675547335726",
    "tag": "0.0.50"
},
```

Second is ok

```
{
    "id": "8ceddb6b6f42dae56e6426b9f865ef4cd5f5c093",
    "message": "async-hooks/use-async-object-manager,linting (bump dependencies versions)",
    "date": "1675547335726",
    "tag": "0.2.261"
},

```

They both have the same datestamp and partly the same message. Looks like something odd happened during tagging (I was using an old version of Bit at the time).  It would be great to know if there is a way to repair this without losing the whole component history.

**Comment by sbland** [comment]
@davidfirst  thanks for taking a look. I'm working across 2 workspaces and can't remember which one it was so checked both. Unfortunately my command history for the first only goes back 2 days (it is also in `./git/bit/command-history` ). It doesn't exist at all in the second workspace. 
It's also worth noting I had an issue where `bvm upgrade` was not updating bit to the latest version (hence version 0.0.744). This was caused by an old Bvm version.
From memory I have only used the following 2 commands for tagging.

- `bit tag --skip-tests -m "commit message`
-  `bit tag -m "commit message`
-  `bit tag --skip-tests  --increment-by=1 -m "commit message"`

Is it possible to fix the bad commit message?

**Comment by davidfirst** [comment]
@sbland , thanks for the info. 
Sadly, there is no easy way I can think of to change the message after it was exported.
Other than the error in bit.cloud when you go to this component history log, are there other issues caused by the wrong format?

**Comment by sbland** [comment]
I cannot view the any of the component pages either locally or on bit.cloud. There doesn't seem to be any other issues. 
Is there a way to reset the component to a previous history point? Currently my only option is to create new components from scratch with new names, then copy the code and then update all dependents.

**Comment by davidfirst** [comment]
> Is there a way to reset the component to a previous history point?
Unfortunately, there is no way to remove exported tags/snaps. I'm thinking to support changing exported tag messages somehow, but not sure it'll happen and when. 

@luvkapur , maybe you can help with the graphql query perspective.
If you go to this page: https://bit.cloud/samnbuk/react_db_client/components/popup-panel-v2 you'll get 500 error because an older version of this component has a corrupted `log.message` property. 
You can easily reproduce it by importing the component and running `bit start`. I'm not sure why all tags history is retrieved when visiting a component home-page and whether is it possible to delay this history until needed.

**Comment by luvkapur** [comment]
> > Is there a way to reset the component to a previous history point?
> 
> Unfortunately, there is no way to remove exported tags/snaps. I'm thinking to support changing exported tag messages somehow, but not sure it'll happen and when.
> 
> @luvkapur , maybe you can help with the graphql query perspective. If you go to this page: https://bit.cloud/samnbuk/react_db_client/components/popup-panel-v2 you'll get 500 error because an older version of this component has a corrupted `log.message` property. You can easily reproduce it by importing the component and running `bit start`. I'm not sure why all tags history is retrieved when visiting a component home-page and whether is it possible to delay this history until needed.

we query logs for the version dropdown, since we need all snaps and tags chronologically. Maybe we can delay querying `log.message` until you go to the changelog or compare (compare needs it for its version selector)

**Comment by sbland** [comment]
It looks like this fix is merged into the latest version and works locally but I'm still getting the error at https://bit.cloud/samnbuk/react_db_client/components/popup-panel-v2. Is there a lag for updating bit.cloud?

**Comment by sbland** [comment]
Additionally I still get the error popup locally but it does not block loading the component.

**Comment by davidfirst** [comment]
Correct. The error is still there, but it doesn't break everything as before.
I checked and your remote-scope has the version 0.1.11 (which I think is the latest stable, not sure).
This fix was introduced on 0.1.19. I believe that the scope will be getting the latest version in the upcoming days.
